llvm/lib/MC/MCAssembler.cpp

```diff
@@ -831,6 +831,20 @@ void MCAssembler::layout(MCAsmLayout &Layout) {
     MCSection *Sec = Layout.getSectionOrder()[i];
     Sec->setLayoutOrder(i);
 
+    // Chain together fragments from all subsections.
```

Comment:
Maybe `if (Sec->Subsections.size() > 1)`? But shouldn't matter much, so feel free to disregard.

Reply:
Since this is not in a bottleneck, I'll keep simple and avoid the condition.